py/server/deephaven/_table_reader.py

```diff
@@ -16,53 +16,77 @@
 
 _JTableUpdateDataReader = jpy.get_type("io.deephaven.integrations.python.PythonTableDataReader")
 
+
 def _col_defs(table: Table, cols: Union[str, Sequence[str]]) -> Sequence[Column]:
     if not cols:
         col_defs = table.columns
     else:
         cols = to_sequence(cols)
         col_defs = [col for col in table.columns if col.name in cols]
+        if len(col_defs) != len(cols):
+            raise ValueError(f"Invalid column names: {set(cols) - {col.name for col in col_defs} }")
 
     return col_defs
 
-def _table_reader_rows(table: Table, cols: Optional[Union[str, Sequence[str]]]) -> Generator[Dict[str, Any], None, None]:
-    """ A generator that reads one row at a time from a table into a dictionary. The dictionary is a map of column names
-    to scalar values of the column data type.
+
+def _table_reader_all_dict(table: Table, cols: Optional[Union[str, Sequence[str]]] = None, *, row_set: jpy.JType,
+                           prev: bool = False, to_numpy: bool = True) -> Dict[str, np.ndarray]:
+    """ Reads all the rows in the given row set of a table into a dictionary. The dictionary is a map of column names
+    to numpy arrays.
```

Comment:
There are a bunch of contradictions in this function signature and related documentation.
1) The docs say that the result is a map of numpy arrays.
2) An argument lets users choose if the output is actually numpy arrays or something else.
3) The type signature assumes that only numpy arrays are returned.